File: 1241-DecompressRunLengthEncodedList/1241-DecompressRunLengthEncodedList.java
// Last updated: 3/26/2026, 1:50:26 AM
class Solution {
    public int[] decompressRLElist(int[] nums) {
        LinkedList<Integer> list = new LinkedList<>();
        for(int i=0;i<nums.length;i+=2){
            int freq = nums[i];
            int value = nums[i+1];
            for(int j=0;j<freq;j++)
                list.add(value);
        }
        int[] result = new int[list.size()];
        for(int i=0;i<list.size();i++)
        {
            result[i] = list.get(i);
        }
        return result;
    }
}

File: 1241-DecompressRunLengthEncodedList/1241-DecompressRunLengthEncodedList.cs
// Last updated: 3/26/2026, 1:50:23 AM
public class Solution {
    public int[] DecompressRLElist(int[] nums) {
        int n = 0;
        
        for(int i=0;i<nums.Length;i=i+2)
            n+=nums[i];
        
        int[] result = new int [n];
        int index = 0;
        
        for(int i=0;i<nums.Length;i=i+2)
        {
            for(int j=0;j<nums[i];j++)
            {
                result[index++] = nums[i+1];
            }
        }
        
        return result;
    }
}